tests/graph_components/validators/test_config_validator.py

```diff
@@ -0,0 +1,805 @@
+            match=(
+                f"You have an overlap between the "
+                f"{RegexEntityExtractorGraphComponent.__name__} and the statistical"
+            ),
+        ):
+            validator.validate(importer)
+    else:
+        with pytest.warns(None) as records:
+            validator.validate(importer)
+        assert len(records) == 0
+
+
+@pytest.mark.parametrize(
+    "component_types_and_configs, should_raise",
+    [
+        (
+            [
+                (
+                    LexicalSyntacticFeaturizerGraphComponent,
+                    {FEATURIZER_CLASS_ALIAS: "different-class-same-name"},
+                ),
+                (
+                    RegexFeaturizerGraphComponent,
+                    {FEATURIZER_CLASS_ALIAS: "different-class-same-name"},
+                ),
+            ],
+            True,
+        ),
+        (
+            [
+                (
+                    RegexFeaturizerGraphComponent,
+                    {FEATURIZER_CLASS_ALIAS: "same-class-other-name"},
+                ),
+                (
+                    RegexFeaturizerGraphComponent,
+                    {FEATURIZER_CLASS_ALIAS: "same-class-different-name"},
+                ),
+            ],
+            False,
+        ),
+        (
+            [(RegexFeaturizerGraphComponent, {}), (RegexFeaturizerGraphComponent, {})],
+            False,
+        ),
+    ],
+)
+def test_nlu_raise_if_featurizers_are_not_compatible(
+    component_types_and_configs: List[Tuple[Type[GraphComponent], Dict[Text, Any]]],
+    should_raise: bool,
+):
+    graph_schema = GraphSchema(
+        {
+            f"{idx}": SchemaNode({}, component_type, "", "", config)
+            for idx, (component_type, config) in enumerate(component_types_and_configs)
+        }
+    )
+    importer = DummyImporter()
+    validator = DefaultV1ConfigValidator(graph_schema)
+    if should_raise:
+        with pytest.raises(InvalidConfigException):
+            validator.validate(importer)
+    else:
+        validator.validate(importer)
+
+
+@pytest.mark.parametrize(
+    "policy_type",
+    [
+        TEDPolicyGraphComponent,
+        RulePolicyGraphComponent,
+        MemoizationPolicyGraphComponent,
+    ],
+)
+def test_core_warn_if_data_but_no_policy(
+    monkeypatch: MonkeyPatch, policy_type: Optional[Type[PolicyGraphComponent]]
+):
+
+    importer = TrainingDataImporter.load_from_dict(
+        domain_path="data/test_e2ebot/domain.yml",
+        training_data_paths=[
+            "data/test_e2ebot/data/nlu.yml",
+            "data/test_e2ebot/data/stories.yml",
+        ],
+    )
+
+    nodes = {"tokenizer": SchemaNode({}, WhitespaceTokenizerGraphComponent, "", "", {})}
+    if policy_type is not None:
+        nodes["some-policy"] = SchemaNode({}, policy_type, "", "", {})
+    graph_schema = GraphSchema(nodes)
+
+    validator = DefaultV1ConfigValidator(graph_schema)
+    monkeypatch.setattr(validator, "_validate_nlu", lambda _: None)
```

Comment:
why do we need the monkey patching? Can we do without it?

Reply:
Some of the monkeypatching is because some of the test ymls have no matching domain file. Shall I relax the checks and just assert a warning is / is not contained in any of the returned user warnings? 
(Current checks assert that the warning is the only one - without the need to check the other warnings explicitly)

Reply:
(This specific patch can be avoided though - just added a DIETClassifier for that - in case that's what you meant)